### ISSUE TITLE
Blueprint static files are not automatically exempted from the rate limiter

### ISSUE THREAD

**Issue by THector99** [alisaifee/flask-limiter#409]
When having a static route in a blueprint, the static files are still rate limited and thus being blocked by Flask limiter

## Expected Behaviour
The files inside the static folder of each blueprint should be exempted from the rate limiter. Every request for a static file should be let trough by flask limter according to the documentation [`The built in flask static files routes are also exempt from rate limits.`](https://flask-limiter.readthedocs.io/en/stable/index.html#:~:text=The%20built%20in%20flask%20static%20files%20routes%20are%20also%20exempt%20from%20rate%20limits.)

## Current Behaviour
The files inside the static folder of each blueprint are not automatically exempted. Thus when the user requests a static file, the request is counted by the rate limter and could be blocked if the rate limit has been exceeded.

## Steps to Reproduce

1. Set global limit to 1 per second (default_limits=["1 per second"],).
2. Have a Flask Blueprint with static files that are requested when loading a page.
3. Load the page, the main html will load but the css and images won't load because of the limiter.
4. Now set the limiter to for example 200 per minute.
5. Reload the page.
6. The assets (css/images etc) now will all show up.

##  Solution
We found a solution that fixes this problem for us based on the following bug reports we could find: [880051368](https://github.com/alisaifee/flask-limiter/issues/285#issuecomment-880051368) and [1491485858](https://github.com/alisaifee/flask-limiter/issues/396#issuecomment-1491485858). We saw that the user in the bug report: [880051368](https://github.com/alisaifee/flask-limiter/issues/285#issuecomment-880051368) had the issue with the main static folder and this was fixed with commit [82bcc28](https://github.com/alisaifee/flask-limiter/commit/82bcc287c5df187b9058cdd608821201a5d245b1). However the latest bug report on this issue ([1491485858](https://github.com/alisaifee/flask-limiter/issues/396#issuecomment-1491485858)) commented that it didn't work with flask blueprints. He had a temporary solution which for us didn't work but we used both of these bug reports to solve the issue for ourselves. In the code below, the code that fixes this issue for us temporarily can be seen.

```
@limiter.request_filter
    def _endpoint_whitelist():
        return "static" in request.endpoint
```

We saw that in commit [82bcc28](https://github.com/alisaifee/flask-limiter/commit/82bcc287c5df187b9058cdd608821201a5d245b1) in file "flask_limiter/extension.py" you added the `request.endpoint == "static"` however when blueprints are used the request.endpoint consists of `blueprint_name.static`. So in order to make it work for blueprints aswell you could change this line: `request.endpoint == "static"` into this: `"static" in request.endpoint`. This will both work when working with or without blueprints. However this will also falsely trigger when "static" is used in the blueprint name. 

## Environment
- Flask-limiter version: 3.3.1
- Flask version: 2.2.3
- Operating sysem: Windows
- Python version: 3.11

**Comment by tassaron** [comment]
In response to the above solution, you could remove false-positives caused by `"static" in request.endpoint` by doing `request.endpoint.split(".")[-1] == "static"`. Maybe I could make a pull request?

**Comment by alisaifee** [comment]
Sorry for the delayed response. `request.endpoint.split(".")[-1] == "static"` works well for all scenarios. This will be fixed in the upcoming release.

**Comment by alisaifee** [comment]
This is now fixed in [3.4.0](https://github.com/alisaifee/flask-limiter/releases/tag/3.4.0)
